### PR TITLE
Apply the featherpad file header and attribution for leafpad

### PR DESCRIPTION
**Pull request by agaida**
fixes tsujan/FeatherPad#68

### CODE REVIEW

**Comment by tsujan** [comment]
Very nice. Thanks!

There's something I don't understand in this and the previous commit. There's no page with the address http://spdx.org/licenses/GPL-3.0+ (or http://spdx.org/licenses/GPL-2.0+). Is that correct? Shouldn't the addressed page exist?

**Comment by tsujan** [comment]
But this address exists: https://opensource.org/licenses/GPL-3.0+

**Comment by tsujan** [comment]
Also this exists: https://spdx.org/licenses/GPL-3.0-or-later.html; but it's strange (the ending "html")

**Comment by agaida** [comment]
[spdx](https://i.imgur.com/YlxzDPm.jpg)
they tricked me :( - last time i looked into the spdx sites it was http:// - so i didn't noticed the addition 's'

**Comment by agaida** [comment]
the or-later is the new official short name for the former +
2.0 or 3.0 will translate to 2.0-only and 3.0-only, what is fine but not widely adopted as the spdx 3.0 was released end of December 17. So for now i would go with the definitions of SPDX 2.X

**Comment by tsujan** [comment]
Sorry! The page exists but I thought the error messages said it didn't.

Nothing important; I'm just curious: What are these error messages:
```
This page contains the following errors:
error on line 176 at column 11: Opening and ending tag mismatch: br line 0 and p
Below is a rendering of the page up to the first error.
```
I can see them both with QupZilla and Chromium. And what's the meaning of "Deprecated"? Does SPDX has problems?

**Comment by agaida** [comment]
> https://spdx.org/licenses/GPL-3.0+.html
no - if you open the just released 3.0 SPDX Definition you will find the "old" deprecated notations like GPL-3.0+ in the appendix.

The problem is that no tool can handle that right now :)

**Comment by agaida** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=887103
i dunno the state of SPDX 3.0 in debian or if the policy team even noticed the release, but SPDX is a safe bet right now i think.

**Comment by tsujan** [comment]
This page is fun;) V3.0 is deprecated since V3.0!!

gpl

Anyhow, not our problem.

**Comment by agaida** [comment]
Read it like
3.0: https://spdx.org/licenses/GPL-3.0-only.html
2.{0,1}: https://spdx.org/licenses/GPL-3.0+.html

**Comment by tsujan** [comment]
@agaida I want to do this before the next release but I also want to know your opinion about it:

If an unusual (and rare) document has a line with more than 500,000 characters, then that line with be truncated at its 500,000th character and this will be appended to it: "HUGE LINE TRUNCATED: NO LINE WITH MORE THAN 500000 CHARACTERS".

My reason is that a document with such lines will take a lot of CPU time to load and I think editing of such lines is senseless. Kate shows a warning message even with 10,000 chars and lets the user decide but if the user chooses to show a lines with 500,000 chars, there will be a lot of CPU usage for several minutes without a response from Kate.

In short, no text will have a line > 500,000 chars.

Does this make sense to you?

**Comment by agaida** [comment]
even 8, 16, 32 or 64k lines are a clean mis-use of an "simple" editor. beyond compare limits lines to 4k lenght in standard. Longer lines are mostly needed if one try to load database dumps and such things. 
hmm, maybe minified js or css - so i would guess 32k are enough, if it is not to much work, make a option for it.

**Comment by agaida** [comment]
500k is more than enough - about 300 sheets of paper :D

**Comment by tsujan** [comment]
Oh, I read your comment again. I think we talked about different things: the number of characters in a single line vs. the number of lines in a single text.

I've already set a limit for file size in FeatherPad but it's huge. I'll reduce it to 100 MiB. Sometimes, I drag and drop PDF and other files into FeatherPad to read hidden info. PDF files are mostly less than 50 MiB but all CPUs can handle 100 MiB, IMO.

For now, I'll let lines with < 50k chars be formatted in HTML/CSS and also will truncate lines with more than 500k -- I know, it's madly huge ;) -- and disable syntax highlighting for texts than contain them altogether. I think this is OK for V0.7. It may be changed later.

Will test the changes for 2 days before adding a commit and will make a new release afterward.

**Comment by tsujan** [comment]
Ah... now, with a big commit -- the last one before the next release -- files > 100 MiB, non-text files and also text file with lines > 500,000 chars can be dropped into FeatherPad without creating any hassle.
